hayeon/programmers-1/비밀지도.js

```diff
@@ -0,0 +1,31 @@
+function solution(n, arr1, arr2) {
+  const result = [];
+
+  const firstMap = arr1.map((el) => el.toString(2).padStart(n, 0));
+  const SecondMap = arr2.map((el) => el.toString(2).padStart(n, 0));
+
+  const fullMap = [];
+
+  for (let i = 0; i < n; i++) {
+    fullMap.push(String(+firstMap[i] + +SecondMap[i]).padStart(n, 0));
+  }
+
+  for (let num of fullMap) {
+    result.push(
+      num
+        .split("")
+        .map((el) => (el === "0" ? (el = " ") : (el = "#")))
+        .join("")
+    );
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(n, arr1, arr2) {
  const firstMap = arr1.map((el) => +el.toString(2));
  const secondMap = arr2.map((el) => +el.toString(2));

  const fullMap = [];

  for (let i = 0; i < n; i++) {
    const unionItems = String(firstMap[i] + secondMap[i])
      .padStart(n, 0)
      .split("");

    fullMap.push(unionItem);
  }

  return fullMap.map((el) => el.map((el) => (el === "0" ? " " : "#")).join(""));
}
```
- 풀이 잘봤습니다 🙇‍♂️ firstMap[i] + secondMap[i]를 더해서 접근하는 방법 너무 좋았습니다 :) 
- 변수명 SecondMap이 아닌 secondMap(카멜 케이스)로 변수명을 작성해주셔야 합니다.
- 하단 for문에서 `padStart()`를 해주기 때문에, firstMap, secondMap에 값을 초기화할 때 사용한 padStart()는 불 필요한 코드입니다 ! 또한 여기서 미리 `+`를 통해 숫자로 형변환을 한다면 밑에 for문에서 복잡도가 낮아질 것 같습니다.
- 첫번째 for문에서 `String(firstMap[i] + SecondMap[i]).padStart(n, 0)` 부분을 변수로 빼서 사용하는게 가독성이 좋아보입니다 ㅎㅎ
- for ... of문을 `map`으로 전환 후에 그 결과를 반환 한다면 result라는 변수와, push와 같은 함수를 사용하지 않으셔도 됩니다 !
- for ... of문 내부 map에서 `el = " "`, `el = "#"` 이렇게 el에다 할당하는 것은 불 필요한 코드입니다. 이렇게 할당하지 않아도 곧바로 리턴이 되기 때문입니다 ~